OpenRA.Game/Activities/Activity.cs

```diff
@@ -50,7 +50,29 @@ public abstract class Activity : IActivityInterface
 	{
 		public ActivityState State { get; private set; }
 
-		protected Activity ChildActivity { get; private set; }
+		Activity childActivity;
+		protected Activity ChildActivity
+		{
+			get
+			{
+				// Never run child activities that are Done. This situation might arise
```

Comment:
We could reduce duplication by defining an `internal Activity NextValidActivity(Activity first)` and calling it with `childActivity` here, `nextActivity` below, and `currentActivity` from `Actor`.